client/state/data-layer/wpcom/sites/stats/google-my-business/test/index.js

```diff
@@ -0,0 +1,97 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import { GOOGLE_MY_BUSINESS_STATS_SEARCH_REQUEST } from 'state/action-types';
+import { fetchGoogleMyBusinessStats, receiveStats } from '../';
+import { receiveGoogleMyBusinessStats } from 'state/google-my-business/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+
+describe( '#fetchGoogleMyBusinessStatsSearch', () => {
+	test( 'should dispatch HTTP request to Google My Business stats endpoint', () => {
+		const action = {
+			type: GOOGLE_MY_BUSINESS_STATS_SEARCH_REQUEST,
+			interval: 'week',
+			aggregation: 'total',
+			siteId: 12345,
+			statType: 'searches',
+		};
+		const dispatch = sinon.spy();
+
+		fetchGoogleMyBusinessStats( { dispatch }, action );
+
+		expect( dispatch ).to.have.been.calledOnce;
+		expect( dispatch ).to.have.been.calledWith(
+			http(
+				{
+					path: '/sites/12345/stats/google-my-business/searches',
+					method: 'GET',
+					query: {
+						interval: 'week',
+						aggregation: 'total',
+					},
+				},
+				action
+			)
+		);
+	} );
+} );
+
+describe( '#receiveStats', () => {
+	test( 'should dispatch receive stats action', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: GOOGLE_MY_BUSINESS_STATS_SEARCH_REQUEST,
+			interval: 'week',
+			aggregation: 'total',
+			siteId: 12345,
+			statType: 'searches',
+		};
+		receiveStats( { dispatch }, action, { hello: 'world' } );
+
+		expect( dispatch ).to.have.been.calledOnce;
+		expect( dispatch ).to.have.been.calledWith(
+			receiveGoogleMyBusinessStats(
+				action.siteId,
+				action.statType,
+				action.interval,
+				action.aggregation,
+				{
+					hello: 'world',
+				}
+			)
+		);
+	} );
+
+	test( 'should transform data snake_case to camelCase', () => {
+		const dispatch = sinon.spy();
+		const action = {
+			type: GOOGLE_MY_BUSINESS_STATS_SEARCH_REQUEST,
+			interval: 'week',
+			aggregation: 'total',
+			siteId: 12345,
+			statType: 'searches',
+		};
+		receiveStats( { dispatch }, action, { hello_world: 'hello' } );
+
+		expect( dispatch ).to.have.been.calledOnce;
+		expect( dispatch ).to.have.been.calledWith(
+			receiveGoogleMyBusinessStats(
+				action.siteId,
+				action.statType,
+				action.interval,
+				action.aggregation,
+				{
+					helloWorld: 'hello',
+				}
```

Comment:
@yurynix: if you had used almost-real data you would have discover that only root property keys are converted to camelCase:

```javascript
{
  "timeZone": "Europe/London",
  "metricValues": [{
    "metric": "QUERIES_DIRECT",
    "total_value": {
      "time_dimension": {
        "timeRange": {
          "endTime": "2018-04-19T23:59:59.900Z",
          "startTime": "2018-04-13T00:00:00Z"
        }
      },
      "metric_option": "AGGREGATED_TOTAL",
      "value": 0
    }
  }, {
    "metric": "QUERIES_INDIRECT",
    "total_value": {
      "time_dimension": {
        "timeRange": {
          "endTime": "2018-04-19T23:59:59.900Z",
          "startTime": "2018-04-13T00:00:00Z"
        }
      },
      "metric_option": "AGGREGATED_TOTAL",
      "value": 1
    }
  }]
}
```

Reply:
Yep, the fact that the server side returned that in camelCase instead of snakeCase masked it even further, great catch @stephanethomas